src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -33,21 +33,15 @@ private static void WorkerThreadStart()
                     {
                         if (TakeActiveRequest())
                         {
+                            var wrapper = ThreadPoolCallbackWrapper.Enter();
                             Volatile.Write(ref ThreadPoolInstance._separated.lastDequeueTime, Environment.TickCount);
                             if (ThreadPoolWorkQueue.Dispatch())
                             {
                                 // If the queue runs out of work for us, we need to update the number of working workers to reflect that we are done working for now
                                 RemoveWorkingWorker();
                             }
 
-                            // Reset thread-local state that we control.
-                            if (currentThread.Priority != ThreadPriority.Normal)
-                            {
-                                currentThread.Priority = ThreadPriority.Normal;
-                            }
-
-                            CultureInfo.CurrentCulture = CultureInfo.InstalledUICulture;
-                            CultureInfo.CurrentUICulture = CultureInfo.InstalledUICulture;
+                            wrapper.Exit(resetThread: true);
```

Comment:
Didn't notice that Dispatch calls ResetThreadPoolThread. We don't need the wrapper at all then since we set up the thread as a thread pool thread when we create it.